pkg/network/encoding/encoding.go

```diff
@@ -52,17 +53,22 @@ func modelConnections(conns *network.Connections) *model.Connections {
 	domainSet := make(map[string]int)
 	routeIndex := make(map[string]RouteIdx)
 	httpIndex := FormatHTTPStats(conns.HTTP)
+	httpMatches := make(map[http.Key]struct{}, len(httpIndex))
 
 	for i, conn := range conns.Conns {
 		httpKey := httpKeyFromConn(conn)
-		agentConns[i] = FormatConnection(conn, domainSet, routeIndex, httpIndex[httpKey])
-		delete(httpIndex, httpKey)
+		httpAggregations := httpIndex[httpKey]
+		if httpAggregations != nil {
+			httpMatches[httpKey] = struct{}{}
+		}
+
+		agentConns[i] = FormatConnection(conn, domainSet, routeIndex, httpAggregations)
 	}
 
-	if len(httpIndex) > 0 {
+	if orphans := len(httpIndex) - len(httpMatches); orphans > 0 {
```

Comment:
If this is the only use for `httpMatches`, could it be just make it an integer counter instead of a map?

Reply:
we can't do that because localhost traffic would be counted twice and external traffic would be counted once